src/nyc_trees/js/src/surveyPage.js

```diff
@@ -288,3 +298,55 @@ function submitSurveyWithTrees() {
         });
     }
 }
+
+$(dom.quit).on('click', quitSurvey);
+
+function quitSurvey() {
+    // Disable submit button to prevent double POSTs
+    $(dom.quit).off('click', quitSurvey);
+
+    var data = createSurveyData();
+    data.survey.has_trees = false;
+    data.survey.quit_reason = $(dom.quitReason).val();
+
+    // There are two views we could POST to, 'survey' and
+    // 'survey_from_event', depending on how we got to this page.
+    //
+    // Both share the same route as the view for this page, so we should be
+    // able to POST to our current URL, whatever it may be (which is the
+    // $.ajax default).
+    $.ajax({
+        type: 'POST',
+        dataType: 'json',
+        data: JSON.stringify(data)
+    })
+    .done(function(content) {
+        $(dom.quitPopup).modal('hide');
+        window.alert('Empty survey saved with quit reason - TODO: show options for what to do next');
+    })
+    .fail(function(jqXHR, textStatus, errorThrown) {
+        toastr.warning('Sorry, there was a problem stopping the survey. Please try again.', 'Something went wrong...');
+    })
+    .always(function() {
+        // Re-enable the submit button
+        $(dom.quit).on('click', quitSurvey);
+    });
+}
+
+$(dom.quitShowPopup).on('click', function(e) {
+    // Bootstrap will handle showing the modal, but the
+    // hidden class, applied by other code, needs to be
+    // removed.
+    $(dom.quitPopup).removeClass('hidden');
+});
+
+// The quit button is not enabled until a quit reason is entered
+$(dom.quit).prop('disabled', true);
+
+$(dom.quitReason).on('change keyup paste', function() {
```

Comment:
The ``'input'`` event does not have the same behavior in a ``textarea``, which is why I ended up with this collection of events suggested after googling the problem.